src/datepicker/datepicker.js

```diff
@@ -629,7 +629,17 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if (!date) {
+              date = new Date(value);
+              if (angular.isDate(date) && !isNaN(date)) {
+                var dateArray = date.toJSON().split('T')[0].split('-');
+                date = new Date(+dateArray[0], +dateArray[1] - 1, +dateArray[2]);
```

Comment:
`date` is already a date instance at this stage - `date = new Date(date)` just breaks the reference and makes a fresh copy, but with the timezone of the previous date object being the key difference. Whether the date passed into the component is from a literal is irrelevant.